pages/post/[id].tsx

```diff
@@ -82,13 +82,22 @@ const Post = ({ post, error }: { post: PostModel, error?: boolean; }) => {
   </>
 }
 
-export const getServerSideProps = async ({ query }) => {
+export const getStaticPaths = async () => {
+  const apiRef = new API();
+  const posts = await apiRef.getPosts();
+  return {
+    paths: posts.map(post => `/post/${post.slug}`),
+    fallback: false
+  }
+}
+
+export const getStaticProps = async ({ params }) => {
   const apiRef = new API();
   try {
-    const post = await apiRef.getPostBySlug(query.id);
-    return { props: { post } };
+    const post = await apiRef.getPostBySlug(params.id);
+    return { props: { post }, revalidate: 1 };
   } catch (e) {
-    return { props: { post: null, error: true } }
+    return { props: { error: true } }
```

Comment:
If the page doesn't have a fallback here, the error handling may not be what you expect

With `fallback` set to `false`, `getStaticProps` will be executed at build time always, instead of handling the error in the UI, you should log the error, and let it throw; that won't affect your users, but the deployment.

If `fallback` is set to `true`, then you can return an error that the UI can handle, but also log it in the console so you can see it in your logs.

Reply:
Good to know. Will update it 👍

Reply:
Fixed, `error` handling removed on the pages, in favor of on `fallback: false`